File: leetcode/python/228_summary_ranges.py
class Solution:
    def summaryRanges(self, nums: list[int]) -> list[str]:
        i = 0
        res: list[str] = []
        while i < len(nums):
            start = end = nums[i]
            while i + 1 < len(nums) and nums[i] == nums[i+1] - 1:
                i += 1
                end = nums[i]
            i += 1
            if start == end:
                res.append(str(start))
            else:
                res.append(f'{start}->{end}')
        return res

s = Solution()
print(s.summaryRanges([0,1,2,4,5,7]))
print(s.summaryRanges([0,2,3,4,6,8,9]))

File: leetcode/rust/src/228_summary_ranges.rs
pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
    let mut ranges: Vec<String> = vec![];
    if nums.is_empty() {
        return ranges;
    }

    let mut l = nums[0];
    let mut r = nums[0];
    for i in 1..nums.len() {
        if nums[i] == r + 1 {
            r = nums[i];
        } else {
            if l == r {
                ranges.push(l.to_string());
            } else {
                ranges.push(format!("{}->{}", l, r));
            }
            l = nums[i];
            r = nums[i];
        }
    }

    if l == r {
        ranges.push(l.to_string());
    } else {
        ranges.push(format!("{}->{}", l, r));
    }


    ranges
}